### ISSUE TITLE
MirroredStrategy AssertionError in 1.11.0-rc0

### ISSUE THREAD

**Issue by cococyx** [tensorflow/tensorflow#22399]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 16.04; Not on mobile device
- **TensorFlow installed from (source or binary)**:
source 
models's commit = 17fa52864bfc7a7444a8b921d8a8eb1669e14ebd and tensorflow's commit = 1e438195399650604fb3aa3a53c67339f1167882
- **TensorFlow version (use command below)**:
git checkout -b v1.11.0-rc0 v1.11.0-rc0
- **Python version**:
Python 3.5.2 (default, Nov 23 2017, 16:37:01)
- **Bazel version (if compiling from source)**:
Build label: 0.17.1
- **GCC/Compiler version (if compiling from source)**:
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
- **CUDA/cuDNN version**:
- **GPU model and memory**:
GeForce GTX 1070 Ti 8G*4
- **Exact command to reproduce**:
# python object_detection/model_main.py --pipeline_config_path=object_detection/samples/configs/ssd_mobilenet_v1_0.75_depth_300x300_coco14_sync.config --logtostderr --model_dir=/data/checkpoint/del2 --num_gpus=4
and meet AssertionError:
File "/home/soft/lib/python2.7/site-packages/tensorflow/contrib/distribute/python/values.py", line 414, in _tensor_conversion_mirrored
    assert not as_ref
AssertionError
You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

### Describe the problem
Similar issue: https://github.com/tensorflow/tensorflow/issues/21968. 
Reply: Actually, this issue is likely fixed in release 1.11, could you try with that and see if that fixes the issue? But I meet the issue in 1.11.0-rc0.

### Source code / logs
 1
git diff object_detection/model_main.py
-config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir)
+distribution = tf.contrib.distribute.MirroredStrategy()
+config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir, train_distribute=distribution, eval_distribute=distribution)
2
cd /models/research
# python object_detection/model_main.py --pipeline_config_path=object_detection/samples/configs/ssd_mobilenet_v1_0.75_depth_300x300_coco14_sync.config --logtostderr --model_dir=/data/sisi.wu/checkpoint/del2 --num_gpus=4
3
[multi-GPU.log](https://github.com/tensorflow/tensorflow/files/2399199/multi-GPU.log)

Thanks.

**Comment by cococyx** [comment]
sorry for missed info:
CUDA/cuDNN version:
Cuda compilation tools, release 9.0, V9.0.176;cuDNN7.0

**Comment by cococyx** [comment]
Update env.txt, [tf_env.txt](https://github.com/tensorflow/tensorflow/files/2399219/tf_env.txt)
Thanks.

**Comment by wt-huang** [comment]
@cococyx Thanks for reporting this. Could you check whether this issue is seen in TensorFlow 1.11.0-rc1?

**Comment by cococyx** [comment]
Thanks for you reply. This issue is seen in TensorFlow 1.11.0-rc1, and I'll try 1.11.0-rc2.
>>> tf.__version__
'1.11.0-rc1'
[multi-GPU-1.11.0-rc1.log](https://github.com/tensorflow/tensorflow/files/2413482/multi-GPU-1.11.0-rc1.log)

**Comment by cococyx** [comment]
>>> tf.__version__
'1.11.0-rc2'
[multi-GPU-1.11.0-rc2.log](https://github.com/tensorflow/tensorflow/files/2413626/multi-GPU-1.11.0-rc2.log)

**Comment by wt-huang** [comment]
@cococyx Thanks for catching and reporting this. It appears there is a sync issue, we will certainly note it down.

**Comment by wt-huang** [comment]
Will close this as the issue is fixed in TensorFlow 1.11.0 and 1.12.0

**Comment by AVCarreiro** [comment]
I still have this issue for Resnet50, and TensorFlow 1.12.0... Seems to be related with tf.slim's convolution2d, but not with batch_norm as I've seen elsewhere: add_weight, instead. 

(On another note, I had problems with batch_norm, namely using sonnet, but disabling the corresponding layers work, so this makes me think that not all layers are supported with MirroredStrategy yet)

**Comment by monckxqq** [comment]
Same error here.
Issue for resnet101 and tensorflow 1.12.0-rc2.
Related with tf.slim's convolution2d  but not with batch_norm.
Have you solve this? @AVCarreiro

**Comment by AVCarreiro** [comment]
Not yet, and it seems more people have this issue.

**Comment by fwu11** [comment]
I had similar error with mirroredstrategy and I am using TensorFlow 1.12.0. I found that if I delete the regularizers.l2_regularizer(weight_decay) in conv2d (set the weight regularizer to None) inside the function 'resnet_arg_scope' in 'resnet_utils' to None then MirroredStrategy works fine.

**Comment by zenglh666** [comment]
> I had similar error with mirroredstrategy and I am using TensorFlow 1.12.0. I found that if I delete the regularizers.l2_regularizer(weight_decay) in conv2d (set the weight regularizer to None) inside the function 'resnet_arg_scope' in 'resnet_utils' to None then MirroredStrategy works fine.

It works for me

**Comment by liyi193328** [comment]
is there any update? same error

**Comment by mihaimaruseac** [comment]
TF 1.11 has passed it's end of life. Please upgrade to TF 1.14 or TF 2.0 beta.

**Comment by dhinkris** [comment]
any update.. I am getting the same error while training based on multi gpu using MirroredStrategy?

**Comment by mihaimaruseac** [comment]
Are you using TF2.0 or TF 1.15?

**Comment by dhinkris** [comment]
TF2.0

**Comment by dhinkris** [comment]
> Are you using TF2.0 or TF 1.15?

I am using TF2.0

**Comment by mihaimaruseac** [comment]
Please open a new issue filling the entire template.

I am going to close this one so there won't be confusion w.r.t. versions used.
